pyeudiw/jwt/__init__.py

```diff
@@ -146,27 +193,18 @@ def decrypt(self, jwe: str) -> dict:
         return msg_dict
 
 
-class JWSHelper:
+class JWSHelper(JWHelperInterface):
     """
     The helper class for work with JWEs.
     """
 
-    def __init__(self, jwk: Union[JWK, dict]):
-        """
-        Creates an instance of JWSHelper.
-
-        :param jwk: The JWK used to sign and verify the content of JWS.
-        :type jwk: Union[JWK, dict]
-        """
-        self.jwk = jwk
-        if isinstance(jwk, dict):
-            self.jwk = JWK(jwk)
-        self.alg = DEFAULT_SIG_KTY_MAP[self.jwk.key.kty]
-
     def sign(
         self,
         plain_dict: Union[dict, str, int, None],
         protected: dict = {},
+        unprotected: dict = {},
+        serialization_format: SerializationFormat = "compact",
```

Comment:
L'uso di JWT in formato JSON (non compatto) con header non protetto è estremamente poco diffuso e non sempre supportato. Molti sviluppatore non sanno nemmeno che esistono. Vale la pena lato nostro supportarli?
La questione è ovviamente analoga lato sd-jwt.

Giro la domanda a @peppelinux per un secondo feedback.

Reply:
purtroppo con eIDAS dobbiamo mantenere la possibilità di usare json serialization

sono d'accordo con te che è meno sicuro di compact serialization, ma eIDAS è eIDAS.